### PR TITLE
Migrate to AWS Elastic Beanstalk

### PR DESCRIPTION
**Pull request by forabi**
* Remove Google App Engine files
* Add the configuration file for AWS CodeBuild (`buildspec.yml`).
* Rewrite `deploy.js` to work with CodeBuild and Travis: it turns out we will have to use Travis for PRs because CodeBuild only supports branch updates. Still, using CodeBuild gives us some nice integration features with AWS services. For example: 
  * we can store the encryption passwords securely in AWS Params Store and use them in CodeBuild to decrypt the secrets
  * we don't have to install the AWS EB CLI tool in the environment image or write our custom logic to bundle and deploy the build files. CodePipeline integrates with CodeBuild to automate this process completelty
  * using CodeBuild allows us to use CodePipeline to stage and review testing versions easily before they are pushed to production

I've configured CodeBuild to build the `master` branch of this repo. CodePipeline will watch the repo, trigger CodeBuild, and deploy the changes automatically to Elastic Beanstalk.

Things that still need to be done:
  * [x] Figure out how to pass branch name to runtime, in order to use it in Sumo Logger.
  * [x] Set up some kind of notification for CodePipeline execution results

### CODE REVIEW

**Comment by forabi** [comment]
@msafih This is ready. Once merged, CodePipeline should automatically deploy to EB.

File: deploy.js [hunk 1]
```diff
@@ -3,59 +3,54 @@
...
+      writeJSONFile('env.json', {
+        BRANCH: CODEBUILD_SOURCE_VERSION,
+        COMMIT_ID: CODEBUILD_RESOLVED_SOURCE_VERSION,
+      }),
+    () => decryptSecrets(secrets, './secrets'),
   ];
 
-  let commands;
+  let isDeployment = false;
   if (isPullRequest === false) {
```
Comment (wholesomedev) [inline deploy.js]: I'm confused by this part. 

```js
if (isPullRequest === false) { console.info('Skipping deployment commands in PRs'); }
```

File: deploy.js [hunk 2]
```diff
@@ -3,59 +3,54 @@
...
-    commands = [...buildCommands, ...deploymentCommands];
-  } else {
-    commands = buildCommands;
     console.info('Skipping deployment commands in PRs');
+  } else if (secrets.some(secret => secret.password === undefined)) {
+    console.info(
+      'Skipping deployment commands because some secrets are not provided',
+    );
+  } else {
+    isDeployment = true;
```
Comment (wholesomedev) [inline deploy.js:43]: If `isPullRequest` is `true` and every secret is defined, then the execution will come to this branch. Is this what we want?

File: server/src/health.ts [hunk 3]
```diff
@@ -1,13 +0,0 @@
...
-
-health.get('/readiness_check', (_, res) => {
-  res.status(200).send('OK');
-});
-
-health.get('/liveness_check', (_, res) => {
-  res.status(200).send('OK');
-});
-
-export { health };
```
Comment (wholesomedev) [inline server/src/health.ts:13]: How did it feel deleting this file? lol

**Review by wholesomedev — CHANGES_REQUESTED** [review]

File: deploy.js (hunk 1 above)
Comment (forabi) [reply deploy.js]: Oops. Fixed.

File: server/src/health.ts (hunk 3 above)
Comment (forabi) [reply server/src/health.ts:13]: Tears of joy

Comment (wholesomedev) [reply server/src/health.ts:13]: 😂

**Review by wholesomedev — APPROVED** [review]

**Pull request merged**